dcpy/models/product/dataset/metadata_v2.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+import jinja2
+from pathlib import Path
+from pydantic import BaseModel
+from typing import Any, List, Literal
+import yaml
+
+from dcpy.utils.logging import logger
+
+
+class CustomizableBase(BaseModel, extra="forbid"):
+    custom: dict[str, Any] | None = None
+
+
+# TODO: move to share with ingest.validate
+class Checks(CustomizableBase):
+    is_primary_key: bool | None
+    nullable: bool | None
+
+
+# TODO: move to share with ingest.validate
+COLUMN_TYPES = Literal["text", "integer", "decimal", "geometry"]
+
+
+class Column(CustomizableBase):
+    id: str
+    display_name: str
+    data_type: COLUMN_TYPES
+
+    data_source: str | None = None
+    description: str | None = None
+    example: str | None = None
+    checks: Checks | None = None
+
+
+class PackageFile(CustomizableBase):
+    id: str
+    filename: str
+
+
+class Package(CustomizableBase):
+    id: str
+    type: str
+    filename: str
+    contains: List[PackageFile]
```

Comment:
nit - maybe `contents` instead of `contains`? Unless there's a specific reason you prefer a verb

Reply:
yeah, `contents` feels nice.